### DISCUSSION TITLE
Mockoon Routing

### DISCUSSION THREAD

**Discussion by AmudalaGirish** [mockoon/mockoon#718] [Help] [answered]
Hi,

We have two routes as follows:

```
1. test/:xxxx
      a. example1 : test/134
      b. example2 : test/172
2. test/:xxxx[:]testName
      a. example1: test/134:testName
      b. example2: test/172:testName
```

I've mocked both the APIs  as shown in  points 1 and 2
Now, when the route "test/134:testName" is hit, it is matching with the first route that is "test/134"
Any thoughts on where I am going wrong?

**Comment by 255kb** [comment, marked as answer]
Hi, i guess you should declare the most specific first, which is the second one. Otherwise, the calls you describe would match the first route too which is z bit more generic.
You can drag and drop route to reorder them.

**Reply by AmudalaGirish** [reply]
Thanks, @255kb that helped.
